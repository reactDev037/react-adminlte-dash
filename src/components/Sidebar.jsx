import React from 'react';

import SidebarMenu from './SidebarMenu';
import SidebarSearch from './SidebarSearch';
import SidebarUser from './SidebarUser';

const Sidebar = ({ children }) => (
  <aside className="main-sidebar">
    {/* sidebar: style can be found in sidebar.less */}
    <section className="sidebar">
      {children}
    </section>
    {/* /.sidebar */}
  </aside>
);

Sidebar.propTypes = {
  children: React.PropTypes.node,
};

Sidebar.Menu = SidebarMenu;
Sidebar.Search = SidebarSearch;
Sidebar.User = SidebarUser;

export default Sidebar;
