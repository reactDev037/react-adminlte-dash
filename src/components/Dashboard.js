/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import styled, { ThemeProvider } from 'styled-components';

import Header from './Header';
import Sidebar from './Sidebar';
import Content from './Content';
import themes from '../styles';

import {
  boxedLayoutMaxWidth,
} from '../styles/variables';

const StyledDashboard = styled.div`
  /* clearfix */
  &:before, &:after {
    display: table;
    content: " ";
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
  }
  &:after {
    clear: both;
  }

  /* theme */
  ${props => (props.theme.sidebarBg && `background-color: ${props.theme.sidebarBg};`)}

  min-height: 100%;
  position: relative;
  overflow: hidden;

  ${props => (props.boxed && `
    max-width: ${boxedLayoutMaxWidth};
    margin: 0 auto;
    min-height: 100%;
    box-shadow: 0 0 8px rgba(0, 0, 0, 0.5);
    position: relative;
  `)}
`;

class Dashboard extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      sidebarCollapse: false,
    };
    this._sidebarToggle = this.sidebarToggle.bind(this);
  }

  sidebarToggle() {
    this.setState({
      sidebarCollapse: !this.state.sidebarCollapse,
    });
  }

  render() {
    const theme = themes[this.props.theme];
    return (
      <StyledDashboard>
        <ThemeProvider theme={theme}>
          <Header
            logoHref={this.props.logoHref}
            logoLg={this.props.logoLg}
            logoSm={this.props.logoSm}
            fixed={this.props.fixed}
            sidebarToggle={this._sidebarToggle}
            sidebarCollapse={this.state.sidebarCollapse}
            sidebarMini={this.props.sidebarMini}
          >
            {this.props.navbarChildren}
          </Header>
        </ThemeProvider>
        <ThemeProvider theme={theme}>
          <Sidebar
            fixed={this.props.fixed}
            sidebarCollapse={this.state.sidebarCollapse}
            sidebarMini={this.props.sidebarMini}
          >
            {this.props.sidebarChildren}
          </Sidebar>
        </ThemeProvider>
        <ThemeProvider theme={theme}>
          <Content
            fixed={this.props.fixed}
            sidebarCollapse={this.state.sidebarCollapse}
            sidebarMini={this.props.sidebarMini}
          >
            {this.props.children}
          </Content>
        </ThemeProvider>
      </StyledDashboard>
    );
  }
}

Dashboard.propTypes = {
  children: React.PropTypes.node,
  logoHref: React.PropTypes.element,
  logoLg: React.PropTypes.element,
  logoSm: React.PropTypes.element,
  navbarChildren: React.PropTypes.node,
  sidebarChildren: React.PropTypes.node,
  fixed: React.PropTypes.bool,
  sidebarMini: React.PropTypes.bool,
  theme: React.PropTypes.string,
};

Dashboard.defaultProps = {
  fixed: false,
  sidebarMini: false,
  theme: 'skin-blue',
};

export default Dashboard;
