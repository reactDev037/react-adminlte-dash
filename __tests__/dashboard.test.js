/* eslint-env jest */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Dashboard from '../src/components/Dashboard';
import Sidebar from '../src/components/Sidebar';

describe('<Dashboard />', () => {
  it('renders correctly with no options', () => {
    const wrapper = mount(<Dashboard />);
    expect(toJson(wrapper)).toMatchSnapshot();

    // Default options
    expect(wrapper.state('sidebarCollapse')).toEqual(false);
    expect(wrapper.prop('initialCollapse')).toEqual(false);
    expect(wrapper.props().fixed).toEqual(false);
    expect(wrapper.props().initialCollapse).toEqual(false);
    expect(wrapper.props().theme).toEqual('skin-blue');
  });

  it('renders correctly with fixed option', () => {
    const component = renderer.create(<Dashboard fixed />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with sidebarMini option', () => {
    const component = renderer.create(<Dashboard sidebarMini />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with fixed and sidebarMini options', () => {
    const component = renderer.create(<Dashboard fixed sidebarMini />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with initialCollapse option', () => {
    const wrapper = mount(<Dashboard initialCollapse />);
    expect(wrapper.state('sidebarCollapse')).toEqual(true);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with initialCollapse and sidebarMini options', () => {
    const component = renderer.create(<Dashboard initialCollapse sidebarMini />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with fixed, initialCollapse, and sidebarMini options', () => {
    const component = renderer.create(<Dashboard fixed initialCollapse sidebarMini />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with black theme', () => {
    const component = renderer.create(<Dashboard theme="skin-black" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with black-light theme', () => {
    const component = renderer.create(<Dashboard theme="skin-black-light" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with blue theme', () => {
    const component = renderer.create(<Dashboard theme="skin-blue" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with blue-light theme', () => {
    const component = renderer.create(<Dashboard theme="skin-blue-light" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with green theme', () => {
    const component = renderer.create(<Dashboard theme="skin-green" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with green-light theme', () => {
    const component = renderer.create(<Dashboard theme="skin-green-light" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with purple theme', () => {
    const component = renderer.create(<Dashboard theme="skin-purple" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with purple-light theme', () => {
    const component = renderer.create(<Dashboard theme="skin-purple-light" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with red theme', () => {
    const component = renderer.create(<Dashboard theme="skin-red" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with red-light theme', () => {
    const component = renderer.create(<Dashboard theme="skin-red-light" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with yellow theme', () => {
    const component = renderer.create(<Dashboard theme="skin-yellow" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with yellow-light theme', () => {
    const component = renderer.create(<Dashboard theme="skin-yellow-light" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with given logo href', () => {
    const href = '/some/link';
    const wrapper = mount(<Dashboard logoHref={href} />);
    expect(wrapper.find('[name="navbar-logo"]').props().href).toEqual(href);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with given large logo', () => {
    const logo = <span>Large</span>;
    const wrapper = mount(<Dashboard logoLg={logo} />);
    expect(wrapper.find('[name="navbar-logo-lg"]').children()
      .matchesElement(logo)).toEqual(true);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with given small logo', () => {
    const logo = <span>Small</span>;
    const wrapper = mount(<Dashboard logoSm={logo} />);
    expect(wrapper.find('[name="navbar-logo-sm"]').children()
      .matchesElement(logo)).toEqual(true);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders navbar children correctly', () => {
    const child = <span>A Child</span>;
    const wrapper = mount(<Dashboard navbarChildren={child} />);
    expect(wrapper.find('[name="navbar-menu-wrapper"]').children()
      .matchesElement(child)).toEqual(true);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders sidebar children correctly', () => {
    const child = <Sidebar.Menu header="Child" />;
    const wrapper = mount(<Dashboard sidebarChildren={child} />);
    expect(wrapper.find('[name="sidebar-wrapper"]').children()
      .matchesElement(child)).toEqual(true);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders children correctly', () => {
    const child = <span>A Child</span>;
    const wrapper = mount(<Dashboard>{child}</Dashboard>);
    expect(wrapper.find('[name="content-wrapper"]').children()
      .matchesElement(child)).toEqual(true);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('opens and closes sidebar on sidebar toggle click', () => {
    const wrapper = mount(<Dashboard />);

    // simulate click to open menu
    wrapper.find('[name="sidebar-toggle"]').simulate('click');
    expect(wrapper.state('sidebarCollapse')).toBe(true);
    expect(toJson(wrapper)).toMatchSnapshot();

    // simulate click to close menu
    wrapper.find('[name="sidebar-toggle"]').simulate('click');
    expect(wrapper.state('sidebarCollapse')).toBe(false);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('opens and closes sidebar with sidebarMini on sidebar toggle click', () => {
    const wrapper = mount(<Dashboard sidebarMini />);

    // simulate click to open menu
    wrapper.find('[name="sidebar-toggle"]').simulate('click');
    expect(wrapper.state('sidebarCollapse')).toBe(true);
    expect(toJson(wrapper)).toMatchSnapshot();

    // simulate click to close menu
    wrapper.find('[name="sidebar-toggle"]').simulate('click');
    expect(wrapper.state('sidebarCollapse')).toBe(false);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
