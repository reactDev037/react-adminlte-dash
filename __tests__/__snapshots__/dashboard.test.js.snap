exports[`<Dashboard /> renders correctly with black theme 1`] = `
<div
  className="bxkLLE">
  <header
    className="dKNVKA">
    <a
      className="gmgior"
      href="/"
      name="navbar-logo">
      <span
        className="ibQovi"
        href="/"
        name="navbar-logo-lg">
        <span>
          <b>
            Admin
          </b>
          LTE
        </span>
      </span>
      <span
        className="hboVZB"
        href="/"
        name="navbar-logo-sm">
        <span>
          <b>
            A
          </b>
          LT
        </span>
      </span>
    </a>
    <nav
      className="kHaBoX">
      <button
        className="fa fa-bars fIoQlD"
        name="sidebar-toggle"
        onClick={[Function]} />
      <div
        className="cTsgzF">
        <ul
          className="bAXwxe"
          name="navbar-menu-wrapper" />
      </div>
    </nav>
  </header>
  <aside
    className="eMTNN">
    <section
      className="hZXMYD"
      name="sidebar-wrapper" />
  </aside>
  <div
    className="fNyzPn"
    name="content-wrapper" />
</div>
`;

exports[`<Dashboard /> renders correctly with black-light theme 1`] = `
<div
  className="bxkLLE">
  <header
    className="dKNVKA">
    <a
      className="gmgior"
      href="/"
      name="navbar-logo">
      <span
        className="ibQovi"
        href="/"
        name="navbar-logo-lg">
        <span>
          <b>
            Admin
          </b>
          LTE
        </span>
      </span>
      <span
        className="hboVZB"
        href="/"
        name="navbar-logo-sm">
        <span>
          <b>
            A
          </b>
          LT
        </span>
      </span>
    </a>
    <nav
      className="kHaBoX">
      <button
        className="fa fa-bars fIoQlD"
        name="sidebar-toggle"
        onClick={[Function]} />
      <div
        className="cTsgzF">
        <ul
          className="bAXwxe"
          name="navbar-menu-wrapper" />
      </div>
    </nav>
  </header>
  <aside
    className="feLxxV">
    <section
      className="hZXMYD"
      name="sidebar-wrapper" />
  </aside>
  <div
    className="fTNHSC"
    name="content-wrapper" />
</div>
`;

exports[`<Dashboard /> renders correctly with blue theme 1`] = `
<div
  className="bxkLLE">
  <header
    className="Sdzae">
    <a
      className="iZvisw"
      href="/"
      name="navbar-logo">
      <span
        className="ibQovi"
        href="/"
        name="navbar-logo-lg">
        <span>
          <b>
            Admin
          </b>
          LTE
        </span>
      </span>
      <span
        className="hboVZB"
        href="/"
        name="navbar-logo-sm">
        <span>
          <b>
            A
          </b>
          LT
        </span>
      </span>
    </a>
    <nav
      className="cPKxdA">
      <button
        className="fa fa-bars eNPEzI"
        name="sidebar-toggle"
        onClick={[Function]} />
      <div
        className="cTsgzF">
        <ul
          className="bAXwxe"
          name="navbar-menu-wrapper" />
      </div>
    </nav>
  </header>
  <aside
    className="eMTNN">
    <section
      className="hZXMYD"
      name="sidebar-wrapper" />
  </aside>
  <div
    className="fNyzPn"
    name="content-wrapper" />
</div>
`;

exports[`<Dashboard /> renders correctly with blue-light theme 1`] = `
<div
  className="bxkLLE">
  <header
    className="Sdzae">
    <a
      className="iZvisw"
      href="/"
      name="navbar-logo">
      <span
        className="ibQovi"
        href="/"
        name="navbar-logo-lg">
        <span>
          <b>
            Admin
          </b>
          LTE
        </span>
      </span>
      <span
        className="hboVZB"
        href="/"
        name="navbar-logo-sm">
        <span>
          <b>
            A
          </b>
          LT
        </span>
      </span>
    </a>
    <nav
      className="cPKxdA">
      <button
        className="fa fa-bars eNPEzI"
        name="sidebar-toggle"
        onClick={[Function]} />
      <div
        className="cTsgzF">
        <ul
          className="bAXwxe"
          name="navbar-menu-wrapper" />
      </div>
    </nav>
  </header>
  <aside
    className="feLxxV">
    <section
      className="hZXMYD"
      name="sidebar-wrapper" />
  </aside>
  <div
    className="fTNHSC"
    name="content-wrapper" />
</div>
`;

exports[`<Dashboard /> renders correctly with fixed and sidebarMini options 1`] = `
<div
  className="bxkLLE">
  <header
    className="glmcVm">
    <a
      className="iZvisw"
      href="/"
      name="navbar-logo">
      <span
        className="ibQovi"
        href="/"
        name="navbar-logo-lg">
        <span>
          <b>
            Admin
          </b>
          LTE
        </span>
      </span>
      <span
        className="hboVZB"
        href="/"
        name="navbar-logo-sm">
        <span>
          <b>
            A
          </b>
          LT
        </span>
      </span>
    </a>
    <nav
      className="cPKxdA">
      <button
        className="fa fa-bars eNPEzI"
        name="sidebar-toggle"
        onClick={[Function]} />
      <div
        className="cTsgzF">
        <ul
          className="bAXwxe"
          name="navbar-menu-wrapper" />
      </div>
    </nav>
  </header>
  <aside
    className="dJjICI">
    <section
      className="hZXMYD"
      name="sidebar-wrapper" />
  </aside>
  <div
    className="KNTAu"
    name="content-wrapper" />
</div>
`;

exports[`<Dashboard /> renders correctly with fixed option 1`] = `
<div
  className="bxkLLE">
  <header
    className="glmcVm">
    <a
      className="iZvisw"
      href="/"
      name="navbar-logo">
      <span
        className="ibQovi"
        href="/"
        name="navbar-logo-lg">
        <span>
          <b>
            Admin
          </b>
          LTE
        </span>
      </span>
      <span
        className="hboVZB"
        href="/"
        name="navbar-logo-sm">
        <span>
          <b>
            A
          </b>
          LT
        </span>
      </span>
    </a>
    <nav
      className="cPKxdA">
      <button
        className="fa fa-bars eNPEzI"
        name="sidebar-toggle"
        onClick={[Function]} />
      <div
        className="cTsgzF">
        <ul
          className="bAXwxe"
          name="navbar-menu-wrapper" />
      </div>
    </nav>
  </header>
  <aside
    className="dJjICI">
    <section
      className="hZXMYD"
      name="sidebar-wrapper" />
  </aside>
  <div
    className="KNTAu"
    name="content-wrapper" />
</div>
`;

exports[`<Dashboard /> renders correctly with fixed, initialCollapse, and sidebarMini options 1`] = `
<div
  className="bxkLLE">
  <header
    className="glmcVm">
    <a
      className="jpQvr"
      href="/"
      name="navbar-logo">
      <span
        className="gdnkfC"
        href="/"
        name="navbar-logo-lg">
        <span>
          <b>
            Admin
          </b>
          LTE
        </span>
      </span>
      <span
        className="bgyKpT"
        href="/"
        name="navbar-logo-sm">
        <span>
          <b>
            A
          </b>
          LT
        </span>
      </span>
    </a>
    <nav
      className="haKxdF">
      <button
        className="fa fa-bars eNPEzI"
        name="sidebar-toggle"
        onClick={[Function]} />
      <div
        className="cTsgzF">
        <ul
          className="bAXwxe"
          name="navbar-menu-wrapper" />
      </div>
    </nav>
  </header>
  <aside
    className="lgMTFt">
    <section
      className="hZXMYD"
      name="sidebar-wrapper" />
  </aside>
  <div
    className="kvraov"
    name="content-wrapper" />
</div>
`;

exports[`<Dashboard /> renders correctly with green theme 1`] = `
<div
  className="bxkLLE">
  <header
    className="Sdzae">
    <a
      className="iKVAkt"
      href="/"
      name="navbar-logo">
      <span
        className="ibQovi"
        href="/"
        name="navbar-logo-lg">
        <span>
          <b>
            Admin
          </b>
          LTE
        </span>
      </span>
      <span
        className="hboVZB"
        href="/"
        name="navbar-logo-sm">
        <span>
          <b>
            A
          </b>
          LT
        </span>
      </span>
    </a>
    <nav
      className="gcPMli">
      <button
        className="fa fa-bars biMfWR"
        name="sidebar-toggle"
        onClick={[Function]} />
      <div
        className="cTsgzF">
        <ul
          className="bAXwxe"
          name="navbar-menu-wrapper" />
      </div>
    </nav>
  </header>
  <aside
    className="eMTNN">
    <section
      className="hZXMYD"
      name="sidebar-wrapper" />
  </aside>
  <div
    className="fNyzPn"
    name="content-wrapper" />
</div>
`;

exports[`<Dashboard /> renders correctly with green-light theme 1`] = `
<div
  className="bxkLLE">
  <header
    className="Sdzae">
    <a
      className="iKVAkt"
      href="/"
      name="navbar-logo">
      <span
        className="ibQovi"
        href="/"
        name="navbar-logo-lg">
        <span>
          <b>
            Admin
          </b>
          LTE
        </span>
      </span>
      <span
        className="hboVZB"
        href="/"
        name="navbar-logo-sm">
        <span>
          <b>
            A
          </b>
          LT
        </span>
      </span>
    </a>
    <nav
      className="gcPMli">
      <button
        className="fa fa-bars biMfWR"
        name="sidebar-toggle"
        onClick={[Function]} />
      <div
        className="cTsgzF">
        <ul
          className="bAXwxe"
          name="navbar-menu-wrapper" />
      </div>
    </nav>
  </header>
  <aside
    className="feLxxV">
    <section
      className="hZXMYD"
      name="sidebar-wrapper" />
  </aside>
  <div
    className="fTNHSC"
    name="content-wrapper" />
</div>
`;

exports[`<Dashboard /> renders correctly with initialCollapse and sidebarMini options 1`] = `
<div
  className="bxkLLE">
  <header
    className="Sdzae">
    <a
      className="jpQvr"
      href="/"
      name="navbar-logo">
      <span
        className="gdnkfC"
        href="/"
        name="navbar-logo-lg">
        <span>
          <b>
            Admin
          </b>
          LTE
        </span>
      </span>
      <span
        className="bgyKpT"
        href="/"
        name="navbar-logo-sm">
        <span>
          <b>
            A
          </b>
          LT
        </span>
      </span>
    </a>
    <nav
      className="haKxdF">
      <button
        className="fa fa-bars eNPEzI"
        name="sidebar-toggle"
        onClick={[Function]} />
      <div
        className="cTsgzF">
        <ul
          className="bAXwxe"
          name="navbar-menu-wrapper" />
      </div>
    </nav>
  </header>
  <aside
    className="eXBQZb">
    <section
      className="hZXMYD"
      name="sidebar-wrapper" />
  </aside>
  <div
    className="dLNMPq"
    name="content-wrapper" />
</div>
`;

exports[`<Dashboard /> renders correctly with initialCollapse option 1`] = `
<div
  className="bxkLLE">
  <header
    className="Sdzae">
    <a
      className="iZvisw"
      href="/"
      name="navbar-logo">
      <span
        className="ibQovi"
        href="/"
        name="navbar-logo-lg">
        <span>
          <b>
            Admin
          </b>
          LTE
        </span>
      </span>
      <span
        className="hboVZB"
        href="/"
        name="navbar-logo-sm">
        <span>
          <b>
            A
          </b>
          LT
        </span>
      </span>
    </a>
    <nav
      className="cPKxdA">
      <button
        className="fa fa-bars eNPEzI"
        name="sidebar-toggle"
        onClick={[Function]} />
      <div
        className="cTsgzF">
        <ul
          className="bAXwxe"
          name="navbar-menu-wrapper" />
      </div>
    </nav>
  </header>
  <aside
    className="hlQeXc">
    <section
      className="hZXMYD"
      name="sidebar-wrapper" />
  </aside>
  <div
    className="yvELr"
    name="content-wrapper" />
</div>
`;

exports[`<Dashboard /> renders correctly with no options 1`] = `
<div
  className="bxkLLE">
  <header
    className="Sdzae">
    <a
      className="iZvisw"
      href="/"
      name="navbar-logo">
      <span
        className="ibQovi"
        href="/"
        name="navbar-logo-lg">
        <span>
          <b>
            Admin
          </b>
          LTE
        </span>
      </span>
      <span
        className="hboVZB"
        href="/"
        name="navbar-logo-sm">
        <span>
          <b>
            A
          </b>
          LT
        </span>
      </span>
    </a>
    <nav
      className="cPKxdA">
      <button
        className="fa fa-bars eNPEzI"
        name="sidebar-toggle"
        onClick={[Function]} />
      <div
        className="cTsgzF">
        <ul
          className="bAXwxe"
          name="navbar-menu-wrapper" />
      </div>
    </nav>
  </header>
  <aside
    className="eMTNN">
    <section
      className="hZXMYD"
      name="sidebar-wrapper" />
  </aside>
  <div
    className="fNyzPn"
    name="content-wrapper" />
</div>
`;

exports[`<Dashboard /> renders correctly with purple theme 1`] = `
<div
  className="bxkLLE">
  <header
    className="Sdzae">
    <a
      className="eyZVfo"
      href="/"
      name="navbar-logo">
      <span
        className="ibQovi"
        href="/"
        name="navbar-logo-lg">
        <span>
          <b>
            Admin
          </b>
          LTE
        </span>
      </span>
      <span
        className="hboVZB"
        href="/"
        name="navbar-logo-sm">
        <span>
          <b>
            A
          </b>
          LT
        </span>
      </span>
    </a>
    <nav
      className="kwFmkS">
      <button
        className="fa fa-bars btgbpw"
        name="sidebar-toggle"
        onClick={[Function]} />
      <div
        className="cTsgzF">
        <ul
          className="bAXwxe"
          name="navbar-menu-wrapper" />
      </div>
    </nav>
  </header>
  <aside
    className="eMTNN">
    <section
      className="hZXMYD"
      name="sidebar-wrapper" />
  </aside>
  <div
    className="fNyzPn"
    name="content-wrapper" />
</div>
`;

exports[`<Dashboard /> renders correctly with purple-light theme 1`] = `
<div
  className="bxkLLE">
  <header
    className="Sdzae">
    <a
      className="eyZVfo"
      href="/"
      name="navbar-logo">
      <span
        className="ibQovi"
        href="/"
        name="navbar-logo-lg">
        <span>
          <b>
            Admin
          </b>
          LTE
        </span>
      </span>
      <span
        className="hboVZB"
        href="/"
        name="navbar-logo-sm">
        <span>
          <b>
            A
          </b>
          LT
        </span>
      </span>
    </a>
    <nav
      className="kwFmkS">
      <button
        className="fa fa-bars btgbpw"
        name="sidebar-toggle"
        onClick={[Function]} />
      <div
        className="cTsgzF">
        <ul
          className="bAXwxe"
          name="navbar-menu-wrapper" />
      </div>
    </nav>
  </header>
  <aside
    className="feLxxV">
    <section
      className="hZXMYD"
      name="sidebar-wrapper" />
  </aside>
  <div
    className="fTNHSC"
    name="content-wrapper" />
</div>
`;

exports[`<Dashboard /> renders correctly with red theme 1`] = `
<div
  className="bxkLLE">
  <header
    className="Sdzae">
    <a
      className="ZVbxS"
      href="/"
      name="navbar-logo">
      <span
        className="ibQovi"
        href="/"
        name="navbar-logo-lg">
        <span>
          <b>
            Admin
          </b>
          LTE
        </span>
      </span>
      <span
        className="hboVZB"
        href="/"
        name="navbar-logo-sm">
        <span>
          <b>
            A
          </b>
          LT
        </span>
      </span>
    </a>
    <nav
      className="rOyox">
      <button
        className="fa fa-bars jatJFQ"
        name="sidebar-toggle"
        onClick={[Function]} />
      <div
        className="cTsgzF">
        <ul
          className="bAXwxe"
          name="navbar-menu-wrapper" />
      </div>
    </nav>
  </header>
  <aside
    className="eMTNN">
    <section
      className="hZXMYD"
      name="sidebar-wrapper" />
  </aside>
  <div
    className="fNyzPn"
    name="content-wrapper" />
</div>
`;

exports[`<Dashboard /> renders correctly with red-light theme 1`] = `
<div
  className="bxkLLE">
  <header
    className="Sdzae">
    <a
      className="ZVbxS"
      href="/"
      name="navbar-logo">
      <span
        className="ibQovi"
        href="/"
        name="navbar-logo-lg">
        <span>
          <b>
            Admin
          </b>
          LTE
        </span>
      </span>
      <span
        className="hboVZB"
        href="/"
        name="navbar-logo-sm">
        <span>
          <b>
            A
          </b>
          LT
        </span>
      </span>
    </a>
    <nav
      className="rOyox">
      <button
        className="fa fa-bars jatJFQ"
        name="sidebar-toggle"
        onClick={[Function]} />
      <div
        className="cTsgzF">
        <ul
          className="bAXwxe"
          name="navbar-menu-wrapper" />
      </div>
    </nav>
  </header>
  <aside
    className="feLxxV">
    <section
      className="hZXMYD"
      name="sidebar-wrapper" />
  </aside>
  <div
    className="fTNHSC"
    name="content-wrapper" />
</div>
`;

exports[`<Dashboard /> renders correctly with sidebarMini option 1`] = `
<div
  className="bxkLLE">
  <header
    className="Sdzae">
    <a
      className="iZvisw"
      href="/"
      name="navbar-logo">
      <span
        className="ibQovi"
        href="/"
        name="navbar-logo-lg">
        <span>
          <b>
            Admin
          </b>
          LTE
        </span>
      </span>
      <span
        className="hboVZB"
        href="/"
        name="navbar-logo-sm">
        <span>
          <b>
            A
          </b>
          LT
        </span>
      </span>
    </a>
    <nav
      className="cPKxdA">
      <button
        className="fa fa-bars eNPEzI"
        name="sidebar-toggle"
        onClick={[Function]} />
      <div
        className="cTsgzF">
        <ul
          className="bAXwxe"
          name="navbar-menu-wrapper" />
      </div>
    </nav>
  </header>
  <aside
    className="eMTNN">
    <section
      className="hZXMYD"
      name="sidebar-wrapper" />
  </aside>
  <div
    className="fNyzPn"
    name="content-wrapper" />
</div>
`;

exports[`<Dashboard /> renders correctly with yellow theme 1`] = `
<div
  className="bxkLLE">
  <header
    className="Sdzae">
    <a
      className="eecIYN"
      href="/"
      name="navbar-logo">
      <span
        className="ibQovi"
        href="/"
        name="navbar-logo-lg">
        <span>
          <b>
            Admin
          </b>
          LTE
        </span>
      </span>
      <span
        className="hboVZB"
        href="/"
        name="navbar-logo-sm">
        <span>
          <b>
            A
          </b>
          LT
        </span>
      </span>
    </a>
    <nav
      className="fccBMc">
      <button
        className="fa fa-bars gQqvlC"
        name="sidebar-toggle"
        onClick={[Function]} />
      <div
        className="cTsgzF">
        <ul
          className="bAXwxe"
          name="navbar-menu-wrapper" />
      </div>
    </nav>
  </header>
  <aside
    className="eMTNN">
    <section
      className="hZXMYD"
      name="sidebar-wrapper" />
  </aside>
  <div
    className="fNyzPn"
    name="content-wrapper" />
</div>
`;

exports[`<Dashboard /> renders correctly with yellow-light theme 1`] = `
<div
  className="bxkLLE">
  <header
    className="Sdzae">
    <a
      className="eecIYN"
      href="/"
      name="navbar-logo">
      <span
        className="ibQovi"
        href="/"
        name="navbar-logo-lg">
        <span>
          <b>
            Admin
          </b>
          LTE
        </span>
      </span>
      <span
        className="hboVZB"
        href="/"
        name="navbar-logo-sm">
        <span>
          <b>
            A
          </b>
          LT
        </span>
      </span>
    </a>
    <nav
      className="fccBMc">
      <button
        className="fa fa-bars gQqvlC"
        name="sidebar-toggle"
        onClick={[Function]} />
      <div
        className="cTsgzF">
        <ul
          className="bAXwxe"
          name="navbar-menu-wrapper" />
      </div>
    </nav>
  </header>
  <aside
    className="feLxxV">
    <section
      className="hZXMYD"
      name="sidebar-wrapper" />
  </aside>
  <div
    className="fTNHSC"
    name="content-wrapper" />
</div>
`;
